api/ims/ims-api.yaml

```diff
@@ -40,79 +38,138 @@ paths:
         default:
           description: unexpected error
           content:
-            'application/json':
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /email/available:
+    get:
+      description: This endpoint compares user's entered email with existing emails and displays an error if there is the same in DB
+      operationId: emailAvailable
+      parameters:
+        - in: query
+          name: email
+          schema:
+            type: string
+            format: email
+          description: Compared email
+      responses:
+        '200':
+          description: Email is not occupied
+        '409':
+          description: User with this email has already registered
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /users/emailVerificationToken/verify:    
+    get:
+      description: To complete the registration you need to confirm your email with token
+      operationId: emailVerificationToken
+      parameters:
+        - in: query
+          name: token
+          schema:
+            type: string
+          description: Token that you need to complete the registration
+      responses:
+        '200':
+          description: Registration completed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SignupResponseWithToken'
+        default:
+          description: User did not confirm his email
+          content:
+            application/json:
               schema:
                 $ref: '#/components/schemas/Error'
 components:
   schemas:
     SignupRequest:
       type: object
-      required:
-        - firstName
-        - lastName
-        - email
-        - password
       properties:
         firstName:
           type: string
           maxLength: 255
-          description: Users first name
+          description: User's first name
         lastName:
           type: string
           maxLength: 255
-          description: Users last name
+          description: User's last name
         email:
           type: string
           format: email
-          description: Users email
+          description: User's email
         password:
           type: string
           format: password
-          description: Users password
-    LoginRequest:
-      type: object
+          description: User's password
       required:
+        - firstName
+        - lastName
         - email
         - password
+    LoginRequest:
+      type: object
       properties:
         email:
           type: string
           format: email
-          description: Users email
+          description: User's email
         password:
           type: string
           format: password
-          description: Users password
+          description: User's password
+      required:
+        - email
+        - password
+    SignupResponse:
```

Comment:
We don't need such a message. status code 200 is enough to understand that the signup request has been processed successfully.